qaframework-bdd-tests/src/main/java/org/openmrs/contrib/qaframework/page/ManageLocationAttributeTypesPage.java

```diff
@@ -105,12 +112,17 @@ public void cancelPurgeLocation() {
 	}
 
 	public void retireLocation() {
-		clickOn(RETIRE_LOCATION_ICON);
+		clickOnLast(RETIRE_LOCATION_ICON);
```

Comment:
actually in this particular case, clickOnLast makes sense to me since in the locations attributes table, newly created location attributes are stored down the table. `Last in goes to the bottom`

Reply:
What difference exists between using `clickOn()` and `clickOnLast()`?

Reply:
clickOn() justs finds the element and clicks on it (using Selenium's [By](https://www.selenium.dev/selenium/docs/api/java/org/openqa/selenium/By.html) to select the element ofcos) whereas clickOnLast() uses the [getLast()](https://guava.dev/releases/20.0/api/docs/com/google/common/collect/Iterables.html#getLast-java.lang.Iterable-T-) method to iterate through the list and get the last element of the list. 

Reference can be found be at https://github.com/openmrs/openmrs-contrib-qaframework/blob/8d5e5d3a848e2a33ddc6533ccddb53366d03ff89/qaframework-bdd-tests/src/main/java/org/openmrs/contrib/qaframework/helper/Page.java#L224-L230.